code/game/objects/structures/tank_dispenser.dm

```diff
@@ -88,40 +108,43 @@
 
 /obj/structure/dispenser/Topic(href, href_list)
 	if(..())
-		return 1
+		return TRUE
 
 	if(Adjacent(usr))
 		usr.set_machine(src)
+
+		// The oxygen tank button
 		if(href_list["oxygen"])
-			if(oxygentanks > 0)
-				var/obj/item/tank/oxygen/O
-				if(oxytanks.len == oxygentanks)
-					O = oxytanks[1]
-					oxytanks.Remove(O)
-				else
-					O = new /obj/item/tank/oxygen(loc)
-				O.loc = loc
-				to_chat(usr, "<span class='notice'>You take [O] out of [src].</span>")
-				oxygentanks--
-				update_icon()
+			if(!stored_oxygen_tanks.len) // No more tanks in the machine
+				return
+
+			var/obj/item/tank/O = stored_oxygen_tanks[1] // Get the first tank in the list
+			stored_oxygen_tanks.Remove(O) // remove it from the list since we are ejecting the tank
+
+			if(!usr.put_in_hands(O)) // Try to place it in the user's hands first
+				O.forceMove(loc) // If hands are full, place it at the location of the tank dispenser
+			to_chat(usr, "<span class='notice'>You take [O] out of [src].</span>")
+			update_icon()
+
+		// The plasma tank button
 		if(href_list["plasma"])
-			if(plasmatanks > 0)
-				var/obj/item/tank/plasma/P
-				if(platanks.len == plasmatanks)
-					P = platanks[1]
-					platanks.Remove(P)
-				else
-					P = new /obj/item/tank/plasma(loc)
-				P.loc = loc
-				to_chat(usr, "<span class='notice'>You take [P] out of [src].</span>")
-				plasmatanks--
-				update_icon()
+			if(!stored_plasma_tanks.len)
+				return
+
+			var/obj/item/tank/P = stored_plasma_tanks[1]
+			stored_plasma_tanks.Remove(P)
+
+			if(!usr.put_in_hands(P))
```

Comment:
Might be best to make a proc for this since you're doing it twice

Reply:
Good idea